h2o-algos/src/main/java/hex/glrm/GLRM.java

```diff
@@ -1131,13 +1186,29 @@ private Chunk chk_xnew(Chunk[] chks, int c) {
 
           // Gradient wrt x_i is matrix product \grad L_{i,j}(x_i * Y_j, A_{i,j}) * Y_j'
           double[] weight = _lossFunc[j].mlgrad(xy, (int) a[j]);
```

Comment:
If you make `mlgrad` function to also accept the array to fill as an argument, then it would be possible to reuse `weight` as well.